### DISCUSSION TITLE
Why do websites built with Next.js crash when using Google Translate? This includes the official Next.js website https://nextjs.org/docs. When you enable the browser's translation feature and click on any link, it crashes.

### DISCUSSION THREAD

**Discussion by tanjie-ola** [vercel/next.js#66313] [Help]
### Summary

Application error: a client-side exception has occurred (see the browser console for more information).
![image](https://github.com/vercel/next.js/assets/113013739/ac9f7053-aa7f-470c-965c-365048db2a7f)

### Example

https://nextjs.org/docs

**Comment by icyJoseph** [comment]
Doesn't directly crash for me though, but often this issue with Google Translate, happens with React (and other frameworks), because Google Translate, replaces text nodes with a `font` tag and some structure to it, and then when React wants to remove that text node, or update it, it can't find it anymore, and that causes an exception.

Like:

```tsx
import { useReducer } from "react";

export default function App() {
  const [flag, toggle] = useReducer((x) => !x, true);
  return (
    <div className="App">
      <h1>Hello CodeSandbox</h1>
      <h2>Start editing to see some magic happen!</h2>

      <button onClick={toggle}>toggle text</button>

      <br />

      {flag && <>{"tomato"}</>}
    </div>
  );
}
```

>      {flag && "tomato"} // also causes the same error

If you translate the site containing that component, and then toggle the flag, it'll crash. The solution is to avoid those floating bits of text, so that React can refer to the wrapping HTML node, like a span:

```tsx

import { useReducer } from "react";

export default function App() {
  const [flag, toggle] = useReducer((x) => !x, true);
  return (
    <div className="App">
      <h1>Hello CodeSandbox</h1>
      <h2>Start editing to see some magic happen!</h2>

      <button onClick={toggle}>toggle text</button>

      <br />

      {flag && <span>{"tomato"}</span>}
    </div>
  );
}
```

~Could you share some reproduction steps?~

**Reply by icyJoseph** [reply]
Alright, clicking around I managed to reproduce it. It is as explained above, caused by trying to work with text nodes, that Google Translate has removed, and replaced with `font` tags.

<img width="1688" alt="Screenshot 2024-05-29 at 10 13 26" src="https://github.com/vercel/next.js/assets/21013447/43214876-d709-4097-939d-a18c70486569">

Since that site is closed source, I won't bother to go through the sources, but this is something that is fixable, and the people maintaining the site should do.

I think many people, want to see a linting rule that could catch this floating bits of text, but there's no yet, any that I am aware of.

**Reply by icyJoseph** [reply]
Someone wrote about this as well: https://martijnhols.nl/gists/everything-about-google-translate-crashing-react

**Comment by madfcat** [comment]
True. Fixing text nodes wrapped in the fragments by wrapping them in `div` instead solved the problem.

**Comment by SindelarPetr** [comment]
A question, the following code would be okay?

```tsx
import { useReducer } from "react";

export default function App() {
  const [flag, toggle] = useReducer((x) => !x, true);
  return (
    <div className="App">
      <h1>Hello CodeSandbox</h1>
      <h2>Start editing to see some magic happen!</h2>

      <button onClick={toggle}>toggle text</button>

      <br />

      tomato
    </div>
  );
}
```

**Comment by jordanjoecooper** [comment]
It seems likely related to this outstanding chromium issue: [Chromium Issue](https://issues.chromium.org/issues/41407169) . Excellent write up here: [Write Up](https://martijnhols.nl/blog/everything-about-google-translate-crashing-react)
